[BUG] - Immediate crash on launch on 4.17.0 on GrapheneOS

After updating to 4.17.0 (https://github.com/eddyizm/tempus/releases/download/v4.17.0/app-tempus-arm64-v8a-release.apk) the app is failing to launch on GrapheneOS, immediately crashes with a generic crash error message and the following stack trace:


java.lang.RuntimeException: Unable to create service com.cappielloantonio.tempo.service.MediaService
	at android.app.ActivityThread.handleCreateService(ActivityThread.java:5593)
	at android.app.ActivityThread.-$$Nest$mhandleCreateService(Unknown Source:0)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2745)
	at android.os.Handler.dispatchMessage(Handler.java:132)
	at android.os.Looper.dispatchMessage(Looper.java:333)
	at android.os.Looper.loopOnce(Looper.java:263)
	at android.os.Looper.loop(Looper.java:367)
	at android.app.ActivityThread.main(ActivityThread.java:9332)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:566)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:837)
Caused by: java.lang.RuntimeException: Mapping failed for song ID: dy3eKOrEoOoLiJUE5bYfW5
	at com.cappielloantonio.tempo.util.d.w(SourceFile:957)
	at com.cappielloantonio.tempo.util.d.A(SourceFile:19)
	at com.cappielloantonio.tempo.service.MediaService.onCreate(SourceFile:585)
	at android.app.ActivityThread.handleCreateService(ActivityThread.java:5582)
	... 10 more
Caused by: java.lang.NullPointerException: Attempt to read from field 'java.lang.Object N2.c.f' on a null object reference in method 'java.util.HashMap Z2.b.e()'
	at Z2.b.e(SourceFile:21)
	at com.cappielloantonio.tempo.util.l.h(SourceFile:6)
	at com.cappielloantonio.tempo.util.d.r(SourceFile:129)
	at com.cappielloantonio.tempo.util.d.w(SourceFile:9)
	... 13 more



Crash was resolved by completely wiping user data and setting up the navidrome connection again from scratch

--- a/app/src/main/java/com/cappielloantonio/tempo/util/MappingUtil.java
+++ b/app/src/main/java/com/cappielloantonio/tempo/util/MappingUtil.java
@@ -36,7 +36,18 @@ public class MappingUtil {
         ArrayList<MediaItem> mediaItems = new ArrayList<>();
 
         for (int i = 0; i < items.size(); i++) {
-            mediaItems.add(mapMediaItem(items.get(i)));
+            Child item = items.get(i);
+            try {
+                mediaItems.add(mapMediaItem(item));
+            } catch (Exception e) {
+                // A single un-mappable saved song (e.g. a cached queue entry whose
+                // Subsonic data lost fields after a server/library change) must not
+                // abort the whole batch. This runs during play-queue restore in
+                // MediaService.onCreate, so a thrown exception crashed the app on
+                // launch. Skip the bad item and keep the rest. See issue #705.
+                Log.e(TAG, "Skipping un-mappable song in queue restore: "
+                        + (item != null ? item.getId() : "null"), e);
+            }
         }
 
         return mediaItems;

